internal/controller/apiextensions/conditions/conditions.go

```diff
@@ -0,0 +1,224 @@
+package conditions
```

Comment:
The idea of this file: Provide a way to update conditions that will automatically emit events when a condition is new or has changed.

This supports the new use case of function conditions, but could also be used by the claim and xr reconcilers to set conditions and have them only emit events if the condition has changed.